File: Programming_Basics_with_Python/conditional_statements_advanced_exercise/03._new_house.py
flower_type = input().capitalize()
quantity_of_flowers = int(input())
budget = int(input())

rose = 5
dahlias = 3.8
tulips = 2.8
narcissus = 3
gladiolus = 2.5

price = 0
discount = 0

if flower_type == 'Roses':
    if quantity_of_flowers > 80:
        price = rose * quantity_of_flowers
        discount = price * 0.1
        price = price - discount
    else:
        price = rose * quantity_of_flowers
elif flower_type == 'Dahlias':
    if quantity_of_flowers > 90:
        price = dahlias * quantity_of_flowers
        discount = price * 0.15
        price = price - discount
    else:
        price = dahlias * quantity_of_flowers
elif flower_type == 'Tulips':
    if quantity_of_flowers > 80:
        price = tulips * quantity_of_flowers
        discount = price * 0.15
        price -= discount
    else:
        price = tulips * quantity_of_flowers
elif flower_type == 'Narcissus':
    if quantity_of_flowers < 120:
        price = narcissus * quantity_of_flowers
        discount = price * 0.15
        price += discount
    else:
        price = narcissus * quantity_of_flowers

elif flower_type == 'Gladiolus':
    if quantity_of_flowers < 80:
        price = gladiolus * quantity_of_flowers
        discount = price * 0.20
        price += discount
    else:
        price = gladiolus * quantity_of_flowers
if price > budget:
    print(f"Not enough money, you need {(price - budget):.2f} leva more.")
else:
    print(
        f"Hey, you have a great garden with {quantity_of_flowers} {flower_type} and {(budget - price):.2f} leva left.")

File: Programing_Basics_with_JS/conditionalStatementsAdvancedExercise/03.NewHouse.js
function newHouse(input) {
    let typeOfFlowers = input[0];
    let flowersCount = Number(input[1]);
    let budget = Number(input[2]);

    let rosePrice = 5;
    let dahliaPrice = 3.80;
    let tulipPrice = 2.80;
    let narcissusPrice = 3;
    let gladiolusPrice = 2.50;

    let finalPriceForFlowers = 0;
    let consoleMessege = '';

    switch(typeOfFlowers) {
        case 'Roses': finalPriceForFlowers = flowersCount * rosePrice;
            if (flowersCount > 80) {
                finalPriceForFlowers = finalPriceForFlowers * 0.9;
            }; 
            break;
        case 'Dahlias': finalPriceForFlowers = flowersCount * dahliaPrice; 
            if (flowersCount > 90) {
                finalPriceForFlowers = finalPriceForFlowers * 0.85;
            };
            break;
        case 'Tulips': finalPriceForFlowers = flowersCount * tulipPrice; 
            if (flowersCount > 80) {
                finalPriceForFlowers = finalPriceForFlowers * 0.85;
            };
            break;
        case 'Narcissus': finalPriceForFlowers = flowersCount * narcissusPrice; 
            if (flowersCount < 120) {
                finalPriceForFlowers = finalPriceForFlowers * 1.15;
            };
            break;
        case 'Gladiolus': finalPriceForFlowers = flowersCount * gladiolusPrice; 
            if (flowersCount < 80) {
                finalPriceForFlowers = finalPriceForFlowers * 1.20;
            };
            break;
    };

    let moneyDiff = Math.abs(budget - finalPriceForFlowers);

    if (budget >= finalPriceForFlowers) {
        consoleMessege = `Hey, you have a great garden with ${flowersCount} ${typeOfFlowers} and ${moneyDiff.toFixed(2)} leva left.`;
    } else {
        consoleMessege = `Not enough money, you need ${moneyDiff.toFixed(2)} leva more.`;
    }

    console.log(consoleMessege);
}

newHouse(["Roses",
"55",
"250"])

newHouse(["Tulips",
"88",
"260"])

newHouse(["Narcissus",
"119",
"360"])
